google/cloud/functions/internal/framework.h

```diff
@@ -49,10 +45,46 @@ using HttpFunction =
  * @see ParseOptions for more details of the command-line arguments used by this
  *     function.
  */
-int Run(int argc, char const* const argv[], HttpFunction handler) noexcept;
+int Run(int argc, char const* const argv[], UserHttpFunction handler) noexcept;
+
+/**
+ * Run the given function, invoking it to handle Cloud Events.
+ *
+ * Starts a HTTP server at the address and listening endpoint described by
+ * @p argv, invoking @p handler to handle any HTTP request which *MUST* conform
+ * to the Cloud Events [HTTP protocol binding][cloud-events-spec]. Note that we
+ * do not expect the application to use this function directly (it is in
+ * `functions_internal` after all), instead, our build scripts should create a
+ * `main()` function that calls this, as shown in the example:
+ *
+ * @par Example
+ * @code
+ * using ::google::cloud::functions_internal::Run;
+ *
+ * CloudEventFunction FindCloudEventHandler() { return some_user_function; }
```

Comment:
(optional) Consider whether these examples would be clearer to readers if we skipped the "FindFooHandler()" indirection, and simply put the actual function that the user should provide. Like:

```cc
void MyCloudEventHandler(functions::CloudEvent event) {
  ...
}
int main(...) {
  return Run(argc, argv, MyCloudEventHandler);
}
```

I think I would find that clearer, unless there's a reason for the "Find" indirection that I'm overlooking.

Reply:
That seems better, done.